src/zenml/integrations/hyperai/service_connectors/hyperai_service_connector.py

```diff
@@ -0,0 +1,386 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""HyperAI Service Connector.
+
+The HyperAI Service Connector allows authenticating to HyperAI (hyperai.ai)
+GPU equipped instances.
+"""
+import base64
+import tempfile
+from typing import Any, List, Optional
+
+import paramiko
+from pydantic import Field, SecretStr
+
+from zenml.integrations.hyperai import (
+    HYPERAI_CONNECTOR_TYPE,
+    HYPERAI_RESOURCE_TYPE,
+)
+from zenml.logger import get_logger
+from zenml.models import (
+    AuthenticationMethodModel,
+    ResourceTypeModel,
+    ServiceConnectorTypeModel,
+)
+from zenml.service_connectors.service_connector import (
+    AuthenticationConfig,
+    ServiceConnector,
+)
+from zenml.utils.enum_utils import StrEnum
+
+logger = get_logger(__name__)
+
+
+class HyperAICredentials(AuthenticationConfig):
+    """HyperAI client authentication credentials."""
+
+    base64_ssh_key: SecretStr = Field(
+        title="SSH key (base64)",
+    )
+    ssh_passphrase: Optional[SecretStr] = Field(
+        default=None,
+        title="SSH key passphrase",
+    )
+
+
+class HyperAIConfiguration(HyperAICredentials):
+    """HyperAI client configuration."""
+
+    ip_address: str = Field(
```

Comment:
I could change it into `hostname`. That way, the field is equal to how `paramiko` calls it, and it would be agnostic to DNS entries or IP addresses. WDYT?